### ISSUE TITLE
shared decoder embeddings with transformer

### ISSUE THREAD

**Issue by vince62s** [OpenNMT/OpenNMT-py#560]
does not seems to work properly.
even after one epoch the PPL / ACC is not even close to what it should be.

Epoch  1,  5900/   -1; acc:  21.58; ppl: 240.95; 4892 src tok/s; 5314 tgt tok/s;  17926 s elapsed
Train perplexity: 897403
Train accuracy: 11.8359
Loading valid dataset from wmt14.en-de.valid.1.pt, number of examples: 2685
Validation perplexity: 375.645
Validation accuracy: 17.8642

without the option, PPL is around 10 ACC around 56.

**Comment by vince62s** [comment]
actually this is not the only issue (I am getting now PPL 134 without shared embeddings)
@helson73 @taoleicn 
Is the transformer on master working for you ?

**Comment by helson73** [comment]
Previously I reported the same issue. 
First, shared embedding implemented here is slightly different with T2T's.
Also, sharing parameters along different modules are not inherently supported in pytorch, I don's know whether setting `embeddings.lut = generator.linear.weight` is correct. 
See here
https://discuss.pytorch.org/t/how-to-create-model-with-sharing-weight/398

**Comment by srush** [comment]
Hmm, so is the issue Transformer generally, or shared embeddings?
